File: front-end/src/features/employees/types/employee.ts
export type EmployeeStatus = 'Active' | 'Inactive'

export type Employee = {
  id: number
  firstName: string
  lastName: string
  email: string
  phone: string
  department: string
  position: string
  hireDate: string
  status: EmployeeStatus
  salary: number
}

File: application/src/Data/Entities/Employee.cs
namespace EmployeeManagementSystem.Data.Entities;

public class Employee
{
    public int EmployeeId { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public int DepartmentId { get; set; }
    public Department? Department { get; set; }
    public string? Position { get; set; }
    public DateOnly HireDate { get; set; }
    public EmployeeStatus Status { get; set; }
    public decimal Salary { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
